code/infrastructure-api/src/controllers/stacksController.spec.js

```diff
@@ -32,26 +33,20 @@ describe('Stacks controller', () => {
         });
     });
 
-    it('should return 400 for invalid request', () => {
-      const invalidRequest = httpMocks.createRequest({
-        method: 'GET',
-        _validationErrors: [{
-          location: 'body',
-          param: 'category',
-          msg: 'category must be specified',
-        }],
-      });
+    it('should return 400 for invalid request', async () => {
+      delete request.body.category;
+      await Promise.all(stacksController.withCategoryValidator.map(validation => validation.run(request)));
```

Comment:
Not sure that you need Bluebird for this. Node Promises should already have `.all` at least in Node 10+.